src/KafkaNET.Library/Consumers/PartitionTopicInfo.cs

```diff
@@ -232,8 +232,8 @@ public int Add(BufferedMessageSet messages)
 
                 Logger.InfoFormat("{2} : Updating fetch offset = {0} with value = {1}", this.fetchedOffset, offset, this.PartitionId);
                 this.chunkQueue.Add(new FetchedDataChunk(messages, this, this.fetchedOffset));
-                long newOffset = Interlocked.Exchange(ref this.fetchedOffset, offset);
-                Logger.Debug("Updated fetch offset of " + this + " to " + newOffset);
```

Comment:
I identified what looked like a different bug here a few months ago:  I think line 235 (Interlocked.Exchange...) needs to happen before line 234 (this.chunkQueue.Add...).  This puts the this.fetchedOffset value is updated to the new offset before passing it off to the new FetchedDataChunk.  Perhaps that fix can be done in a different PR, but I believe it should be done.
Thanks for making the logging change, though.  That looks good.